### ISSUE TITLE
Profile view doesn't always load new data correctly

### ISSUE THREAD

**Issue by boreq** [planetary-social/nos#371]
Steps:
- open a profile you haven't opened before by pasting the npub into the discover screen input
- profile doesn't load
- go back to discover screen by clicking back
- paste the npub again
- profile loads

This happened for me with `npub1skkpx369ld4z350w4ah8mwrqcsm3kk5udv2qluxcdepftpqvcx4snr68s6` and `rsslay.nos.pink`.

https://github.com/planetary-social/nos/assets/1935975/09baea91-e737-4337-b360-a1c558d34900

**Comment by mplorentz** [comment]
I've made some more optimizations to our relay communication and fixed some bugs. Unfortunately there can still be long delays to load many things, including the profile page. The bottleneck now is that we are just requesting way too many events. For instance on my profile where I follow 100 people Nos had to process 26,000 in the first 2 minutes of runtime. 25,000 of these events were duplicates - we had already parsed them before.

I'll look for any easy ways to cut down on the number of events we are parsing next week, but until we implement the gossip/outbox model the number of events we parse will always be greater than or equal to the number of events we request multiplied by the number of relays a user is using, which can quickly get out of hand.
